fs_storage/models/fs_storage.py

```diff
@@ -0,0 +1,426 @@
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# Copyright 2019 Camptocamp SA (http://www.camptocamp.com).
+# @author Simone Orsi <simone.orsi@camptocamp.com>
+# Copyright 2023 ACSONE SA/NV (https://www.acsone.eu).
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from __future__ import annotations
+
+import base64
+import functools
+import inspect
+import json
+import logging
+import os.path
+import re
+import warnings
+from typing import AnyStr
+
+import fsspec
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import ValidationError
+
+from odoo.addons.base_sparse_field.models.fields import Serialized
+
+_logger = logging.getLogger(__name__)
+
+
+# TODO: useful for the whole OCA?
+def deprecated(reason):
+    """Mark functions or classes as deprecated.
+
+    Emit warning at execution.
+
+    The @deprecated is used with a 'reason'.
+
+        .. code-block:: python
+
+            @deprecated("please, use another function")
+            def old_function(x, y):
+                pass
+    """
+
+    def decorator(func1):
+
+        if inspect.isclass(func1):
+            fmt1 = "Call to deprecated class {name} ({reason})."
+        else:
+            fmt1 = "Call to deprecated function {name} ({reason})."
+
+        @functools.wraps(func1)
+        def new_func1(*args, **kwargs):
+            warnings.simplefilter("always", DeprecationWarning)
+            warnings.warn(
+                fmt1.format(name=func1.__name__, reason=reason),
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
+            warnings.simplefilter("default", DeprecationWarning)
+            return func1(*args, **kwargs)
+
+        return new_func1
+
+    return decorator
+
+
+class FSStorage(models.Model):
+    _name = "fs.storage"
+    _backend_name = "fs_storage"
```

Comment:
I did not see it used anymore, is it still usefull ?

Reply:
Good catch